web/tools.html

```diff
@@ -131,61 +135,66 @@ <h4>Tools</h4>
-                                manifestType = "Type: " + manifestType.replaceAll("\"","");
-
-                                let manifestItemCount = "Number of Items: " + loadedManifest.items.length;
-
-                                loadMessage.innerHTML = "<u>Current Object:</u>";
-                                //This should also be replaced with separate columns, in case there may be longer fields to display.
-                                manifestLabelField.innerHTML = manifestLabel + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + manifestType + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + manifestItemCount;
-                            }
-                            catch{
-                                loadMessage.innerHTML = "No metadata available!";
-                                manifestLabelField.innerHTML = "";
-                            }
-                        })
-                        .catch(error => {
-                            manifestMessage.textContent = 'Failed to load manifest. Please check the URL and try again.';
-                            manifestMessage.style.color = 'red';
-                            console.error('Error:', error);
-                        });
+                    showLoading();
+
+                    try {
+                        const response = await fetch(url);
+                        if (!response.ok) {
+                            throw new Error(`HTTP error! status: ${response.status}`);
+                        }
+                        const data = await response.json();
+            
+                        // Process the successfully loaded manifest
+                        hideLoading();
+                        manifestMessage.textContent = 'Manifest loaded successfully!';
+                        manifestMessage.style.color = 'green';
+
+                        storeManifestLink(url);
+            
+                        // Process manifest metadata
+                        try {
+                            let manifestLabel = JSON.stringify(data.label.en[0]);
+                            manifestLabel = "Name: " + manifestLabel.replaceAll("\"","");
+
+                            let manifestType = JSON.stringify(data.type);
+                            manifestType = "Type: " + manifestType.replaceAll("\"","");
+
+                            let manifestItemCount = "Number of Items: " + data.items.length;
+
+                            loadMessage.innerHTML = "<u>Current Object:</u>";
+                            manifestLabelField.innerHTML = manifestLabel + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + 
+                                            manifestType + "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;" + 
+                                            manifestItemCount;
+                        } catch (metadataError) {
+                            loadMessage.innerHTML = "No metadata available!";
```

Comment:
I feel like we could make the fallback message a little more informative, maybe like "Metadata not available for this manifest"